applications/openshift/kubelet/kubelet_protect_kernel_defaults/rule.yml

```diff
@@ -0,0 +1,41 @@
+documentation_complete: true
+
+prodtype: ocp4
+
+title: 'Kubelet - Disable Kernel Default Protection'
+
+description: |-
+    By default, the kubelet can adjust kernel parameters automatically.
+    If kernel default protection is enabled, the kubelet will instead
+    error if kernel parameters do not match what it expects.
+
+rationale: |-
+    OpenShift tunes the kernel and kubelet configurations dynamically. If
+    kernel default protection is enabled, changes could lead to the kubelet
+    crash-looping.
```

Comment:
Okay, but what is the security rationale? So, for example, if the kubelet can change any kernel parameter, that means an attacker can figure out a way to disable kernel host protections through the kubelet.

Reply:
What we want to check for is that the kubelet _can_ adjust kernel parameters  - if this option is turned on for the kubelet, then a change in the kernel's parameters (not by the kubelet) can crash-loop the kubelet. The security rational in this case would be to avoid a DoS of the kubelet.

Reply:
This is actually a bad thing if I am understanding you correctly. If the kubelet can can adjust kernel parameters, it means the kubelet can be used to attack a node, and an attacker can figure out how to hijack a kubelet to change kernel parameters to enable more insecure settings or disable secure settings.

Reply:
Possibly, but that's not the intent of the protection recommendation in the draft. If it is turned on you probably won't have a running kubelet.

Reply:
Let's change the rationale to explain to folks that not allowing the kubelet to tune kernel parameters could lead to the kubelet itself crashing and you losing availability in your deployment. Kernel parameters could tuned in such a way as to restrict a container even further... and when running a container that does security scanning (such as sysdig) you'd need to relax some tunables as well (thinking about sysctl's)

Reply:
Utlimately, this needs to be fixed in OCP. A node should be flagged when kernel tunables are disabled to OCP so that it doesn't continue to be polled, and an alternative method through machineconfigs should set the tunables vs the kubelet doing it. As of this point, the security risk in OCP is for enabling or disabling this component. You enable it, you risk DOS and kubelet problems because of a couple of sysctls not set. You disable it, you risk node security by allowing a kubelet configure kernel parameters. As a point of reference, this is not a problem for other kube distros. For more security conscious customers and requirements, this is going to have to be fixed.